EC-218: Basket revision (#1215)
ezsystems/developer-documentation: docs/guide/basket/basket_configuration.md

# Basket configuration

## Additional data in the basket line

Each basket item can contain an additional line of data.

![](../img/basket_additional_data_1.png)

> You can enable this additional line and set its length in **Configuration Settings** in the **eCommerce** tab in the backend:

Author: ```suggestion
You can enable this additional line and set its length in the Back Office, the **eCommerce** tab, **Configuration Settings**:
```

> You can enable this additional line and set its length in **Configuration Settings** in the **eCommerce** tab in the backend:

Author: Also a tab, so it would look sound clumsy to repeat the word.

```

## Basket storage time

The time for which a basket is stored depends on whether the basket belongs to an anonymous user or a logged-in user.

A basket for a logged-in customer is stored forever.

A basket for an anonymous user is stored for 120 hours by default.
You can configure a different value:

``` yaml
ses_basket.default.validHours: 120
```

You can use the `silversolutions:baskets:clear` command to delete anonymous expired baskets:

``` bash
php bin/console silversolutions:baskets:clear <validHours>
```

It deletes all anonymous baskets from the database that are older than `validHours`.

For example:

``` bash
php bin/console silversolutions:baskets:clear 720
```

## Discontinued products

A listener can check if a product is still available, or discontinued.
You can disable this setting in configuration:

``` yaml
siso_basket.default.discontinued_products_listener_active: false
```

The listener checks if the current stock is greater than or equal to the quantity the customer wants to order.
In this case the order is allowed.

The optional setting `discontinued_products_listener_consider_packaging_unit` enables ignoring the packaging unit

> in order to sell the remaining products even if the remaining stock does not fit the packing unit rule

Reviewer: ```suggestion
in order to sell the remaining products, even if the remaining stock does not fit the packing unit rule
```

> (e.g. packing unit contains 10 pieces but 9 are left in stock).

Reviewer: ```suggestion
(for example, the packing unit is 10 pieces but 9 are left in stock).
```

> The listener reduces the quantity in the order to the number of products which are in stock. 

Reviewer: ```suggestion
The listener reduces the quantity in the order to the number of products that are in stock. 
```
